sdk/ai/azure-ai-inference/src/main/java/com/azure/ai/inference/ChatCompletionClientTracer.java

```diff
@@ -0,0 +1,606 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.inference;
+
+import com.azure.ai.inference.models.ChatChoice;
+import com.azure.ai.inference.models.ChatCompletions;
+import com.azure.ai.inference.models.ChatCompletionsOptions;
+import com.azure.ai.inference.models.ChatCompletionsToolCall;
+import com.azure.ai.inference.models.ChatRequestMessage;
+import com.azure.ai.inference.models.ChatRole;
+import com.azure.ai.inference.models.CompletionsFinishReason;
+import com.azure.ai.inference.models.CompletionsUsage;
+import com.azure.ai.inference.models.StreamingChatChoiceUpdate;
+import com.azure.ai.inference.models.StreamingChatCompletionsUpdate;
+import com.azure.ai.inference.models.StreamingChatResponseMessageUpdate;
+import com.azure.ai.inference.models.StreamingChatResponseToolCallUpdate;
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.ClientOptions;
+import com.azure.core.util.Context;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.TracingOptions;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.tracing.SpanKind;
+import com.azure.core.util.tracing.StartSpanOptions;
+import com.azure.core.util.tracing.Tracer;
+import com.azure.core.util.tracing.TracerProvider;
+import com.azure.json.JsonProviders;
+import com.azure.json.JsonWriter;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayDeque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringJoiner;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Tracing for the convenience methods in {@link com.azure.ai.inference.ChatCompletionsClient} and
+ * {@link com.azure.ai.inference.ChatCompletionsAsyncClient}.
+ */
+final class ChatCompletionClientTracer {
+    private static final String INFERENCE_GEN_AI_SYSTEM_NAME = "az.ai.inference";
+    private static final String OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT
+        = "OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT";
+    private static final String FINISH_REASON_ERROR = "{\"finish_reason\": \"error\"}";
+    private static final String FINISH_REASON_CANCELED = "{\"finish_reason\": \"canceled\"}";
```

Comment:
Is there any reason to use `canceled` for `finish_reason` and `cancelled` for `error.type`? 
We use `cancelled` error type in EH, SB, and in logging, so I wonder if we could change `finish_reason` to match it.

Reply:
Good point, updated to use `cancelled` so we’re consistent.